src/test/java/com/google/sps/DatastoreMetadataStoreTest.java

```diff
@@ -0,0 +1,220 @@
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+
+import com.google.sps.servlets.Metadata;
+import com.google.sps.servlets.DatastoreMetadataStore;
+
+public class DatastoreMetadataStoreTest {
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  // Entities to be put into [mock] Datastore
+  private Entity datasetEntity1;
+  private Entity datasetEntity2;
+  private Entity datasetEntity3;
+  private Entity datasetEntity4;
+  private Entity datasetEntity5;
+  private Entity datasetEntity6;
+  private Entity datasetEntity7;
+
+  // Sample metadata information for an entry in Datastore
+  private static final String USER = "johndallard@google.com";
+  private static final String DATASET = "my_test_dataset";
+  private static final String MODEL = "DELG";
+  private static final String VISUALIZATION = "t-SNE";
+  private static final int NUMBER_OF_IMAGES = 7;
+  private static final long TIMESTAMP = 1596730599767L;
+
+  // Instance of DatastoreMetadatastore class 
+  private MetadataStore datastoreStorage = new DatastoreMetadataStore();
+  
+  @Before
+  public void setUp() {
+    helper.setUp();
+    setUpDatastore();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+  
+  @Test
+  public void basicMetadataRetrieval() {
+    // Test to show correct metadata retrieval for a particular dataset
+
+    Metadata metadata = Metadata.of(USER, DATASET, MODEL, VISUALIZATION, NUMBER_OF_IMAGES, TIMESTAMP);
+    datastoreStorage.storeData(metadata);
+    Assert.assertEquals(metadata, datastoreStorage.retrieveMetadata(metadata.dataset()));
+  }
+  
+  @Test
+  public void retrieveNonexistantMetadata() {
+    // Test to show what happens when a dataset doesn't exist
+
+    Metadata metadata = datastoreStorage.retrieveMetadata("non_existant_dataset");
+    Assert.assertEquals(null, metadata);
+  }
+  
+  @Test
+  public void testRetrieveUsersDatasets() {
+    // Test the proper retrieval of a user's datasets
+
+    List<String> testDatasetList = new ArrayList<String>();
+    testDatasetList.add("johnnys_first_dataset");
+    testDatasetList.add("test_testerson");
+    testDatasetList.add("my_set");
+    testDatasetList.add("latest_and_greatest");
+    testDatasetList.add("set_five");
+    testDatasetList.add("test_dataset");
+
+    Assert.assertEquals(testDatasetList, datastoreStorage.getUsersDatasets("johndallard@google.com"));
```

Comment:
You can do:
 assertThat(datastoreStorage.getUserDatasets(..)).containsExactly("johnnys_first...", ...)